packages/common-grpc/src/service.js

```diff
@@ -886,7 +768,143 @@ GrpcService.prototype.getService_ = function(protoOpts) {
   return service;
 };
 
+/**
+ * A class that can be used to convert an object to a struct. Optionally this
+ * class can be used to erase/throw on circular references during conversion.
+ *
+ * @private
+ *
+ * @param {object=} options - Configuration object.
+ * @param {boolean} options.removeCircular - Remove circular references in the
+ *     object with a placeholder string. (Default: `false`)
+ * @param {boolean} options.stringify - Stringify un-recognized types. (Default:
+ *     `false`)
+ */
+function ObjectToStructConverter(options) {
+  options = options || {};
+
+  this.seenObjects = new Set();
+  this.removeCircular = options.removeCircular === true;
+  this.stringify = options.stringify === true;
+}
+
+/**
+ * Begin the conversion process from a JS object to an encoded gRPC Value
+ * message.
+ *
+ * @param {*} value - The input value.
+ * @return {object} - The encoded value.
+ *
+ * @example
+ * ObjectToStructConverter.convert({
+ *   aString: 'Hi'
+ * });
+ * // {
+ * //   fields: {
+ * //     aString: {
+ * //       stringValue: 'Hello!'
+ * //     }
+ * //   }
+ * // }
+ */
+ObjectToStructConverter.prototype.convert = function(obj) {
+  var convertedObject = {
+    fields: {}
+  };
+
+  this.seenObjects.add(obj);
+
+  for (var prop in obj) {
+    if (obj.hasOwnProperty(prop)) {
+      var value = obj[prop];
+
+      if (is.undefined(value)) {
+        continue;
+      }
+
+      convertedObject.fields[prop] = this.encodeValue_(value);
+    }
+  }
+
+  this.seenObjects.clear();
```

Comment:
Yes, thanks for catching. I put it in before I noticed that `encodeValue_` calls `convert`. Changed it back.